lib/bucket_files.js

```diff
@@ -30,15 +30,32 @@ module.exports = (client, bucket, logger) => {
 
   const current = async () => {
     try {
-      logger.info(`executing listObjects for bucket: ${bucket}`)
-      const results = await client.listObjects({ Bucket: bucket }).promise()
-      logger.debug(`listObjects response for bucket: ${bucket}`, results)
-      logger.info(`listObjects returned ${results.Contents.length} for bucket: ${bucket}`)
-      return results.Contents
+        var params = { Bucket: bucket };
+        var allObjectsRead = false;
+        var objectsRead = new Array();
+        logger.info(`executing listObjects for bucket: ${bucket}`);
+        while (!allObjectsRead) {
+            var results = await client.listObjectsV2(params).promise();
+            results.Contents.forEach(element => {
+                // Could delete non-essential properties from the data here
+                // For example: element.StorageClass = null ;
+                //
+                // Could lower the memory footprint by converting the Timestamp to UTC millis
+                //              element.LastModified = new Date(element.LastModified).getTime();
+                objectsRead.push(element);
+            });
+            if (results.IsTruncated)
```

Comment:
Can you add brackets around this statement? if () {} else {} to match the style of the existing code.